app/src/main/java/com/codingblocks/cbonlineapp/mycourse/MyCourseViewModel.kt

```diff
@@ -202,6 +204,28 @@ class MyCourseViewModel(
             dm.enqueue(request)
         }
     }
+
+
+    fun getPerformance() {
+        runIO {
+            when (val response = repo.getPerformance()) {
+                is ResultWrapper.GenericError -> setError(response.error)
+                is ResultWrapper.Success -> with(response.value) {
+                    if (isSuccessful) {
+                        body()?.let { response ->
+                            repo.saveRank(response)
```

Comment:
This would add a new hb everytime ?

Reply:
Yes, Bhaiya.  Actually what I thought is to add rank in DB every time when user start an app.  But these approach is adding rank every time when the user navigates to Overview Fragment.
Will this work or should I try something else?

Reply:
Will it work if I make current rank as Primary Key?, So that it won't add same rank every time

Reply:
Even in overview fragment it would n enteries for opening up app n times and making the rank as the primary key still in that case there be multiple enteries